### PR TITLE
Change job data serialization format to EDN

### PR DESCRIPTION
**Pull request by jimbru**
JSON is lossy with respect to Clojure data. Instead let's use EDN. See commit log for more detail.

### CODE REVIEW

File: resources/migrations/102-edn.up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- WARNING: This migration will nuke all in-flight jobs!
+
+DELETE FROM backtick_queue;
```
Comment (ctdean) [inline resources/migrations/102-edn.up.sql:3]: Boom!  Good idea,

File: src/backtick/engine.clj [hunk 2]
```diff
@@ -32,7 +33,7 @@
   (db/queue-insert! {:name name
                      :priority (to-sql-time (time/now))
                      :state "queued"
-                     :data [data]}))
+                     :data (prn-str data)}))
```
Comment (ctdean) [inline src/backtick/engine.clj:37]: I'm assuming you checked that the (A) matches my comment B below.

File: src/backtick/engine.clj [hunk 3]
```diff
@@ -116,7 +130,7 @@
             (db/queue-insert! {:name (:name payload)
                                :state "running"
                                :priority (to-sql-time (time/now))
-                               :data [[]]})
+                               :data (prn-str [[]])})
```
Comment (ctdean) [inline src/backtick/engine.clj]: And here (B), I assume you checked that the data structure we're storing is the same for a cron job and a regular job with no arguments .

**Comment by ctdean** [comment]
LGTM - assuming you've done the checks mentioned in the comments.

+1

File: src/backtick/engine.clj (hunk 3 above)
Comment (jimbru) [inline src/backtick/engine.clj]: Yeah that's the hope... Is there any way we can exercise this code path in a test?

**Pull request merged**